### PR TITLE
Update useSearchModal logic

### PR DESCRIPTION
**Pull request by KaemonIsland**
## Hey, I just made a Pull Request!

I noticed that there was an issue with the `useSearchModal` logic.

`toggleModal` was hardcoded to keep the modal open.

`setOpen` would keep the modal open, if it was already open.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/search/src/components/SearchModal/useSearchModal.tsx [hunk 1]
```diff
@@ -32,19 +32,21 @@ export function useSearchModal(initialState = false) {
     open: initialState,
   });
 
+  // Toggle the current state
   const toggleModal = useCallback(
     () =>
       setState(prevState => ({
-        open: true,
+        open: !prevState.open,
```
Comment (camilaibs) [inline plugins/search/src/components/SearchModal/useSearchModal.tsx:39]: @vinzscam, if I'm not confused, you said this state was on purpose, could you review this change, please?

Comment (vinzscam) [reply plugins/search/src/components/SearchModal/useSearchModal.tsx:39]: Hi @KaemonIsland, 👋  thanks for looking into this.
As @camilaibs mentioned the logic here is correct.

There are some tests cases which describe this specific behaviour, for example [here](https://github.com/backstage/backstage/blob/master/plugins/search/src/components/SearchModal/useSearchModal.test.tsx#L58). So changing this behavior will cause some tests `useSearchModal.test.tsx` to fail.

The reason behind this is that we don't want the modal to be unmounted, once it has been opened.
Applying your logic, will cause the modal and the `SearchContext` to be unmounted, meaning that any state will be lost and a new fetch request will be sent to the server.

But I am curious about your use case: is this causing any problem in your application?

**Comment by freben** [comment]
aside from the comments given by others: any time that you make an API-affecting change, you'll have to run `yarn build:api-reports` in the root and commit the results as well (see the CI build output)

File: plugins/search/src/components/SearchModal/useSearchModal.tsx (hunk 1 above)
Comment (KaemonIsland) [reply plugins/search/src/components/SearchModal/useSearchModal.tsx:39]: Ah, I see what you mean, and I understand why I was confused.

The issue was that our `SidebarSearchModal` would open but not close. (We created searchModal). The SearchModal uses an `open` prop, and I had passed `state.open` to it when I should have passed `state.hidden`.

That makes sense why it was hardcoded to true now.

I'm used to `open` being the variable name to determine if a modal should be displayed or not.

I'll close this!

Comment (vinzscam) [reply plugins/search/src/components/SearchModal/useSearchModal.tsx:39]: ah I see! That's why [here](https://github.com/backstage/backstage/blob/master/plugins/search/src/components/SidebarSearchModal/SidebarSearchModal.tsx#L38) we spread the state.

**Pull request closed without merging**